edlib/src/edlib.cpp

```diff
@@ -605,7 +609,8 @@ static int myersCalcEditDistanceSemiGlobal(
             lastBlock++; bl++; Peq_c++;
             bl->P = static_cast<Word>(-1); // All 1s
             bl->M = static_cast<Word>(0);
-            bl->score = (bl - 1)->score - hout + WORD_SIZE + calculateBlock(bl->P, bl->M, *Peq_c, hout, bl->P, bl->M);
+            pair<Word, Word> PMhout = calculateBlock(bl->P, bl->M, *Peq_c, Phout, Mhout, bl->P, bl->M);
+            bl->score = (bl - 1)->score - hout + WORD_SIZE + PMhout.first - PMhout.second;
```

Comment:
If we apply the changes I suggested above, then we also need to adjust this part a bit:

```suggestion
            pair<Word, Word> nextPMhout = calculateBlock(bl->P, bl->M, *Peq_c, PMhout.first, PMhout.second, bl->P, bl->M);
            const int nextHout = nextPMhout.first - nextPMhout.second;
            bl->score = (bl - 1)->score - hout + WORD_SIZE + nextHout;
```